jgrapht-core/src/main/java/org/jgrapht/graph/interval/Interval.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * (C) Copyright 2003-2018, by Daniel Mock and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph.interval;
+
+import java.util.Objects;
+
+/**
+ * Model of an interval in the interval graph.
```

Comment:
The Javadoc should make clear the semantics of the interval bounds (open/closed).